ValveResourceFormat/Resource/ResourceTypes/ModelAnimation/AnimationSegmentDecoder.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+
+namespace ValveResourceFormat.ResourceTypes.ModelAnimation
+{
+    public class AnimationSegmentDecoder
+    {
+        public int[] Elements { get; }
+        public AnimationDataChannel LocalChannel { get; }
+
+        public AnimationSegmentDecoder(int[] elements, AnimationDataChannel localChannel)
+        {
+            Elements = elements;
+            LocalChannel = localChannel;
+        }
+
+        public virtual void Read(int frameIndex, Frame outFrame)
+        {
+            throw new NotImplementedException();
```

Comment:
Why throw at runtime when you can just make this abstract and error at compile time